### PR TITLE
Upgrade To `actix-web` v1.0.0

### PR DESCRIPTION
**Pull request by nathankleyn**
Update `actix-web` requirement from `^0.7.18` to `^1.0.0`.

There were significant breaking changes described in the migration guide
at https://github.com/actix/actix-web/blob/master/MIGRATION.md#100 — and
a lot that weren't documented anywhere!

Some notable ones are:

* A new way to get at query and path params (which is much nicer,
admittedly).
* A completely new way to specify routes and middleware — the DSL is
basically entirely overhauled.
* Removal of the `TestServer` helpers from `actix-web` means you now
need to test through `actix-http-test` instead.
* We are now able to assert on the body of failed requests, meaning the
tests can check the actual error responses on those too.

### CODE REVIEW

File: server/src/main.rs [hunk 1]
```diff
@@ -173,25 +155,26 @@ mod tests {
         run_server(DEFAULT_HOST, DEFAULT_PORT, Path::new(""));
     }
 
-    fn start_test_server() -> TestServer {
-        TestServer::build_with_state(|| {
+    fn start_test_server() -> TestServerRuntime {
+        TestServer::new(|| {
```
Comment (nathankleyn) [inline server/src/main.rs]: This is now coming from `actix-http-test`, so...

File: server/src/main.rs [hunk 2]
```diff
@@ -173,25 +155,26 @@ mod tests {
...
             let addr = GitRepos::new(git::load_repos(Path::new("test"))).start();
-            AppState {
-                git_repos: addr.clone(),
-            }
-        })
-        .start(|app| {
-            app.resource("/repo/{repo}", |r| r.h(&super::get_repo));
+
+            h1::H1Service::new(
```
Comment (nathankleyn) [inline server/src/main.rs]: Expects an `actix-http` service to be passed, rather than the old API which expected `actix-web` types. This is fine, they are compatible anyway — it's just a shuffle around, really.

File: server/src/main.rs [hunk 3]
```diff
@@ -173,25 +155,26 @@ mod tests {
...
     }
 
     macro_rules! assert_test_server_responds_with {
         ($path:expr, $expected_status:expr, $expected_body:expr) => {{
             let mut srv = start_test_server();
 
-            let request = srv.client(http::Method::GET, &$path).finish().unwrap();
-            let response = srv.execute(request.send()).unwrap();
-            let bytes = srv.execute(response.body()).unwrap();
+            let mut response = srv.block_on(srv.get(&$path).send()).unwrap();
```
Comment (nathankleyn) [inline server/src/main.rs]: The `TestServer` API completely changed.

File: server/src/main.rs [hunk 4]
```diff
@@ -204,21 +187,25 @@ mod tests {
         assert_test_server_responds_with!(
             "/repo/idontexist?file=README.md&reference=origin/master",
             404,
-            ""
+            "No repo found with name 'idontexist'"
```
Comment (nathankleyn) [inline server/src/main.rs]: We now get error responses back too, finally — you can see there's a few other tests below where I now assert error responses (not just status) now. 🎉

File: server/src/main.rs [hunk 5]
```diff
@@ -204,21 +187,25 @@ mod tests {
...
+        )
     }
 
     #[test]
     fn get_repo_with_invalid_file() {
         assert_test_server_responds_with!(
             "/repo/fixtures?file=not-a-file&reference=origin/master",
             404,
-            ""
+            "the path 'not-a-file' does not exist in the given tree; class=Tree (14); code=NotFound (-3)"
```
Comment (nathankleyn) [inline server/src/main.rs]: I would like to fix these next two error messages to not be directly `git2`'s wording — but job for another PR! 🤷‍♂
